kratos/solving_strategies/strategies/explicit_solving_strategy.h

```diff
@@ -0,0 +1,651 @@
+     */
+    virtual bool SolveSolutionStep()
+    {
+        // Call the initialize non-linear iteration
+        InitializeNonLinearIterationContainer(GetModelPart().Elements());
+        InitializeNonLinearIterationContainer(GetModelPart().Conditions());
+        InitializeNonLinearIterationContainer(GetModelPart().MasterSlaveConstraints());
+
+        // Solve the problem assuming that a lumped mass matrix is used
+        SolveWithLumpedMassMatrix();
+
+        // If required, update the mesh with the obtained solution
+        if (mMoveMeshFlag) {
+            MoveMesh();
+        }
+
+        // Call the finalize non-linear iteration
+        FinalizeNonLinearIterationContainer(GetModelPart().Elements());
+        FinalizeNonLinearIterationContainer(GetModelPart().Conditions());
+        FinalizeNonLinearIterationContainer(GetModelPart().MasterSlaveConstraints());
+
+        return true;
+    }
+
+    /**
+     * @brief This sets the level of echo for the solving strategy
+     * @param Level of echo for the solving strategy
+     * @details
+     * {
+     * 0 -> Mute... no echo at all
+     * 1 -> Printing time and basic informations
+     * 2 -> Printing advanced information
+     * 3 -> Print of debug informations: Echo of stiffness matrix, Dx, b...
+     * }
+     */
+    virtual void SetEchoLevel(const int Level)
+    {
+        mEchoLevel = Level;
+    }
+
+    /**
+     * @brief This returns the level of echo for the solving strategy
+     * @details
+     * {
+     * 0 -> Mute... no echo at all
+     * 1 -> Printing time and basic informations
+     * 2 -> Printing advanced information
+     * 3 -> Print of debug informations: Echo of stiffness matrix, Dx, b...
+     * }
+     * @return Level of echo for the solving strategy
+     */
+    int GetEchoLevel() const
+    {
+        return mEchoLevel;
+    }
+
+    /**
+     * This sets the build level
+     * @param Level The build level
+     * @details
+     * {
+     * 0 -> Set up the DOF set just once
+     * 1 -> Set up the DOF set at the beginning of each solution step
+     * }
+     */
+    void SetRebuildLevel(int Level)
+    {
+        mRebuildLevel = Level;
+    }
+
+    /**
+     * @brief This returns the build level
+     * @details
+     * {
+     * 0 -> Set up the DOF set just once
+     * 1 -> Set up the DOF set at the beginning of each solution step
+     * }
+     * @return The build level
+     */
+    int GetRebuildLevel() const
+    {
+        return mRebuildLevel;
+    }
+
+    /**
+     * @brief This function sets the flag that says if the mesh is moved
+     * @param Flag True if the mesh is moved, false otherwise
+     */
+    void SetMoveMeshFlag(bool Flag)
+    {
+        mMoveMeshFlag = Flag;
+    }
+
+    /**
+     * @brief This function returns the flag that says if the mesh is moved
+     * @return True if the mesh is moved, false otherwise
+     */
+    bool MoveMeshFlag()
```

Comment:
```suggestion
    bool MoveMeshFlag() const
```

I would at least make it const